Detectors/TRD/workflow/src/TRDGlobalTrackingSpec.cxx

```diff
@@ -0,0 +1,201 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   TRDGlobalTrackingSpec.cxx
+
+#include "TRDWorkflow/TRDGlobalTrackingSpec.h"
+
+#include "TRDBase/Geometry.h"
+
+#include "DetectorsCommonDataFormats/NameConf.h"
+#include "CommonConstants/LHCConstants.h"
+#include "DetectorsBase/GeometryManager.h"
+#include "DetectorsBase/Propagator.h"
+#include "ReconstructionDataFormats/TrackTPCITS.h"
+#include "DataFormatsTRD/Tracklet64.h"
+#include "DataFormatsTRD/TriggerRecord.h"
+
+// GPU header
+#include "GPUReconstruction.h"
+#include "GPUChainTracking.h"
+#include "GPUSettings.h"
+#include "GPUDataTypes.h"
+#include "GPUTRDDef.h"
+#include "GPUTRDTrack.h"
+#include "GPUTRDTrackletWord.h"
+#include "GPUTRDInterfaces.h"
+#include "GPUTRDGeometry.h"
+
+using namespace o2::framework;
+using namespace o2::gpu;
+
+namespace o2
+{
+namespace trd
+{
+
+void TRDGlobalTracking::init(InitContext& ic)
+{
+  //-------- init geometry and field --------//
+  o2::base::GeometryManager::loadGeometry();
+  o2::base::Propagator::initFieldFromGRP(o2::base::NameConf::getGRPFileName());
+  auto geo = Geometry::instance();
+  geo->createPadPlaneArray();
+  geo->createClusterMatrixArray();
+  const GeometryFlat geoFlat(*geo);
+
+  //-------- init GPU reconstruction --------//
+  GPUSettingsEvent cfgEvent;                 // defaults should be ok
+  GPUSettingsRec cfgRec;                     // don't care for now, NWaysOuter is set in here for instance
+  GPUSettingsProcessing cfgDeviceProcessing; // also keep defaults here, or adjust debug level
+  cfgDeviceProcessing.debugLevel = 5;
```

Comment:
I would not use debug level 5 for production code.

Reply:
I am using now the default values as you suggested

Reply:
Nevermind, simply the default values won't work. I need to set the GPURecoStepConfiguration at least, because otherwise I get the following errors:
[205400:trd-globaltracking]: [14:33:32][ERROR] Invalid input, TPC Clusterizer needs TPC raw input
[205400:trd-globaltracking]: [14:33:32][FATAL] GPUReconstruction could not be initialized

Reply:
@martenole : Indeed you obviously need to set the recosteps. Can you retry with #4813 which has the correct setter for it :)

Reply:
Thanks for adding that setting, but I would keep it as it is for now, so that I can easily adjust debug level or other settings during the development. I guess as soon as the tracklets are ready I think this will be needed..